docs/_ext/autoclass_with_examples.py

```diff
@@ -0,0 +1,116 @@
+import re
+import sys
+from typing import Any, Dict, List, Tuple
+
+from docutils.nodes import Node
+from sphinx.ext.autodoc.directive import AutodocDirective
+
+from starknet_py.constants import ROOT_PATH
+
+
+class AutoclassWithExamples(AutodocDirective):
+    """
+    Custom extension of the AutodocDirective class, which is used to
+    add code examples to the method docstrings in the documentation.
+
+    This class runs before the AutodocDirective class and pulls code snippets
+    from the starknet_py/tests/e2e/docs/code_examples directory to include in the documentation.
+    This allows developers to easily see and understand how the methods
+    being documented are intended to be used in a practical context.
+
+    .. note::
+
+        The docstrings will not be inherited if the hierarchy is `deeper` than one step.
```

Comment:
Let's create an issue for that.

Reply:
https://github.com/software-mansion/starknet.py/issues/705